lib/string.js

```diff
@@ -91,19 +91,34 @@ internals.String = class extends Any {
         });
     }
 
-    regex(pattern, name) {
+    regex(pattern, patternOptions) {
 
         Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
 
-        pattern = new RegExp(pattern.source, pattern.ignoreCase ? 'i' : undefined);         // Future version should break this and forbid unsupported regex flags
+        const patternObject = {
+            pattern: new RegExp(pattern.source, pattern.ignoreCase ? 'i' : undefined)         // Future version should break this and forbid unsupported regex flags
+        };
+
+        const patternIsInverted = (typeof patternOptions === 'object' && patternOptions.invert);
```

Comment:
I was actually just taking a second look at that. The stringified piece is just because I copied the Lab test output. But I am seeing this:
```js
const schema = Joi.string().regex(/[a-z]/, {
    invert: true
});
const description = schema.describe();
console.log(description.rules[0]);
/*
{ name: 'regex',
  arg: { pattern: /[a-z]/, name: undefined, invert: true } }
*/
```
If having `name: undefined` is okay, then no problem. I just wouldn't expect any property at all in this case.